static/app/components/replays/highlightReplayPlugin.tsx

```diff
@@ -9,77 +9,148 @@ interface HighlightReplayPluginOptions {
+      return;
+    }
 
-      const {top, left, width, height} = node.getBoundingClientRect();
-      const highlightColor = highlightObj.color ?? this.defaultHighlightColor;
+    replayer.wrapper.removeChild(highlightObj.canvas);
+    this.highlightsByNodeId.delete(nodeId);
+  }
 
-      // Clone canvas
-      const canvas = mouseTail.cloneNode();
+  handler(event: eventWithTime, _isSync: boolean, {replayer}: {replayer: Replayer}) {
+    // Check for currently hightlighted nodes that have been removed from the DOM
+    if (
+      this.highlightsByNodeId.size &&
+      'removes' in event.data &&
+      event.data.removes?.length
+    ) {
+      // If a highlighted node was removed from DOM we need to remove the highlight canvas
+      // const highlightedNodeIds = Array.from(this.highlightsByNodeId.keys());
+      event.data.removes.forEach(removedNode => {
+        if ('id' in removedNode && this.highlightsByNodeId.has(removedNode.id)) {
+          this.removeHighlight(removedNode.id, replayer);
+        }
+      });
+    }
 
-      const ctx = canvas.getContext('2d');
+    // Check for expired highlights
+    if (this.highlightsByNodeId.size) {
+      Array.from(this.highlightsByNodeId.entries())
+        .filter(
+          ([, {added, expires}]) => event.timestamp < added || event.timestamp >= expires
+        )
+        .forEach(([nodeId]) => {
+          this.removeHighlight(nodeId, replayer);
+        });
+    }
 
-      if (!ctx) {
-        return;
-      }
+    // If we have highlights and the replay event is a scroll event, update the highlight
+    // `event.type` == EventType.IncrementalSnapshot (https://github.com/rrweb-io/rrweb/blob/master/packages/rrweb/src/types.ts#L20)
+    // `event.data.source` == IncrementalSource.Scroll (https://github.com/rrweb-io/rrweb/blob/master/packages/rrweb/src/types.ts#L83)
+    if (this.highlightsByNodeId.size && event.type === 3 && event.data.source === 3) {
+      // TODO(replays): Re-draw rect instead of removing it (:
+      Array.from(this.highlightsByNodeId.entries()).forEach(([nodeId]) => {
+        this.removeHighlight(nodeId, replayer);
+      });
+    }
 
-      // TODO(replays): Does not account for scrolling (should we attempt to keep highlight visible, or does it disappear)
+    if (!isHighlightEvent(event) || event.data.nodeId < 0) {
+      return;
+    }
 
-      // Draw a rectangle to highlight element
-      ctx.fillStyle = highlightColor;
-      ctx.fillRect(left, top, width, height);
+    const highlightObj = event.data;
 
-      // Draw a dashed border around highlight
-      ctx.beginPath();
-      ctx.setLineDash([5, 5]);
-      ctx.moveTo(left, top);
-      ctx.lineTo(left + width, top);
-      ctx.lineTo(left + width, top + height);
-      ctx.lineTo(left, top + height);
-      ctx.closePath();
-      ctx.stroke();
+    // @ts-expect-error mouseTail is private
+    const {mouseTail, wrapper} = replayer;
```

Comment:
Lets stick a comment here or below at the `mouseTail.clone()` call explaining that we want to clone the canvas. 

Looks like it is so we get the class/width/height attrs attached. We want our canvas to be positioned properly overtop of the capture, just like the mouseTail is.